### PR TITLE
Upgrade to Ruby 3.3.2

### PR DESCRIPTION
**Pull request by nimmolo**
Ruby is currently at 3.3.6, with 3.4 on the way. I figure minor version upgrades are the way to go, one by one.

So i'm planning to do this, then 3.3.4, then 3.3.6 in as short a time as possible.

### CODE REVIEW

**Review by mo-nathan — APPROVED** [review]
Got it all working and the tests are passing for me locally.

**Comment by nimmolo** [comment]
@mo-nathan You mentioned this bumping Rails to 7.1.5, but i'm not seeing that in `Gemfile.lock`.

**Comment by mo-nathan** [comment]
I'm honestly not sure what happened there.  I must have been moving too quickly with too many other things going on.

File: Gemfile.lock [hunk 1]
```diff
@@ -573,7 +573,7 @@ DEPENDENCIES
   xmlrpc
 
 RUBY VERSION
-   ruby 3.3.0p0
+   ruby 3.3.2p78
 
 BUNDLED WITH
    2.5.18
```
Comment (JoeCohen) [inline Gemfile.lock:579]: Should bundler also be updated? Latest version is 2.6.2. https://rubygems.org/gems/bundler/versions

**Review by JoeCohen — APPROVED** [review]

**Comment by mo-nathan** [comment]
I think this should get closed in favor of #2602.

**Pull request closed without merging**